scripts/changeset/dependabot-changeset.sh

```diff
@@ -1,17 +1,27 @@
 #!/bin/bash
 
-# Get the latest commit hash on the current branch
-latest_commit=$(git rev-parse HEAD)
+# Get the first dependabot commit which contains the changed files
+first_commit=$(git log master..HEAD --oneline --reverse | xargs | awk '{print $1;}')
 
-# Get the list of changed package directories
-changed_packages=$(git diff-tree --no-commit-id --name-only -r "$latest_commit" | grep "packages/" | sed -E 's|packages/([^/]+)/.*|\1|')
+changed_files=$(git diff-tree --no-commit-id --name-only -r "$first_commit")
+
+# Get the list of relevant package.json changes
+changed_packages=$(echo "$changed_files" | grep -E "packages/.+/package.json" | xargs jq -r '.name')
+changed_templates=$(echo "$changed_files" | grep -E "templates/.+/package.json")
```

Comment:
Let's log these out as well.

```suggestion
changed_packages=$(echo "$changed_files" | grep -E "packages/.+/package.json" | xargs jq -r '.name')
changed_templates=$(echo "$changed_files" | grep -E "templates/.+/package.json")

echo "changed packages: $changed_packages"
echo "changed templates: $changed_templates"
```